WooCommerce/Classes/ViewRelated/Reviews/ReviewsViewModel.swift

```diff
@@ -5,8 +5,41 @@ import Yosemite
 
 import class AutomatticTracks.CrashLogging
 
+protocol ReviewsViewModelOutput {
+    var isEmpty: Bool { get }
 
-final class ReviewsViewModel {
+    var dataSource: UITableViewDataSource { get }
+
+    var delegate: ReviewsInteractionDelegate { get }
+
+    var hasUnreadNotifications: Bool { get }
+
+    var shouldPromptForAppReview: Bool { get }
+
+    var hasErrorLoadingData: Bool { get set }
+
+    func containsMorePages(_ highestVisibleReview: Int) -> Bool
+}
+
+protocol ReviewsViewModelActionsHandler {
+    func displayPlaceholderReviews(tableView: UITableView)
+
+    func removePlaceholderReviews(tableView: UITableView)
+
+    func configureResultsController(tableView: UITableView)
+
+    func refreshResults()
+
+    func configureTableViewCells(tableView: UITableView)
+
+    func markAllAsRead(onCompletion: @escaping (Error?) -> Void)
+
+    func synchronizeReviews(pageNumber: Int,
+                            pageSize: Int,
+                            onCompletion: (() -> Void)?)
+}
+
+final class ReviewsViewModel: ReviewsViewModelOutput, ReviewsViewModelActionsHandler {
```

Comment:
Another nit: we may need a comment saying this is the view model for the ReviewsViewController?

Reply:
Done in https://github.com/woocommerce/woocommerce-ios/pull/5834/commits/dc81e07adf5f5e269b86dc319ee2c291d53ef5fc